### PR TITLE
Add variables `TRT02A`, `TRT02P`, `TRT02SDTM`, `TRT02EDTM`, `AP02SDTM`, `AP02EDTM`, `TRT01SDTM`, `TRT01EDTM`,    `AP01SDTM`,and `AP01EDTM` to `adsl`

### LINKED ISSUE #192: Add 1 more period TRT02A into ADSL
### Feature description

One more period TRT02A is required in ADSL
In DSUR4 TLGC: https://code.roche.com/nest/docs/tlg-catalog/devel/-/merge_requests/130

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by JiaLiu0001**
close #192

### CODE REVIEW

**Comment by shajoezhu** [comment]
Hi Jia, thanks for working on this. I have one minor comment, since this change happens in ADSL, can you investigate if it has impacts downstream in other datasets (it could be none). Thanks!

**Review by codesformochi — COMMENTED** [review]
looks good! there's only one comment from Joe.

**Comment by JiaLiu0001** [comment]
> looks good! there's only one comment from Joe.

Thanks, @shajoezhu @codesformochi . 
Yes exactly, there are some vars needed in downstream datasets. Besides that, some date-related vars also need to be added.
1. Besides TRT02A and TRT02P, the following vars should also need. 
`TRT02SDTM`, `TRT02EDTM`, `TRT01SDTM`, and `TRT01EDTM`;
`AP02SDTM`,  `AP02EDTM`, `AP01SDTM`, and  `AP01EDTM`
2. Downstream-related datasets need to be updated to add new columns above.
3.  TRTSDTM <- TRT01SDTM.  
     TRTEDTM <- TRT02EDTM.

**Comment by shajoezhu** [comment]
hi @JiaLiu0001 , upstream is changed, can you update this please. Thanks!

File: inst/metadata/ADSL.yml [hunk 1]
```diff
@@ -96,6 +102,22 @@ variables:
...
+  TRT02EDTM:
+    label: Datetime of Last Exposure to Treatment in Period 02
+  AP01SDTM:
+    label: Period 01 Start Datetime
+  AP01EDTM:
+    label: Period 01 End Datetime
+  AP02SDTM:
+    label: Period 02 Start Datetime
+  AP02EDTM:
+    label: Period 02 End Datetime
```
Comment (shajoezhu) [inline inst/metadata/ADSL.yml:120]: add type

File: R/radsl.R [hunk 2]
```diff
@@ -87,12 +87,20 @@ radsl <- function(N = 400, # nolint
...
-    st_posixn = as.numeric(.data$TRTSDTM),
-    TRTEDTM = as.POSIXct(.data$st_posixn + study_duration_secs, origin = "1970-01-01"),
+    st_posixn = as.numeric(.data$TRT01SDTM),
+    TRT01EDTM = as.POSIXct(.data$st_posixn + study_duration_secs / 4, origin = "1970-01-01"),
+    AP01EDTM = TRT01EDTM,
+    st_posixn_2 = as.numeric(.data$TRT01EDTM),
+    TRT02SDTM = as.POSIXct(.data$st_posixn_2 + study_duration_secs / 8, origin = "1970-01-01"),
+    AP02SDTM = TRT02SDTM,
+    st_posixn_3 = as.numeric(.data$TRT02SDTM),
+    TRTEDTM = as.POSIXct(.data$st_posixn_3 + study_duration_secs / 2, origin = "1970-01-01"),
```
Comment (shajoezhu) [inline R/radsl.R:103]: parameterize 2, 4, 8

**Review by shajoezhu — CHANGES_REQUESTED** [review]
1. add flag with_trt02, default to NULL, when default, there is no TRT02*, 
2. when with_trt02 is not NULL, start making 
  
TRT01SDTM = TRTSDTM, 
make TRT01EDTM
TRT02SDTM = TRT01EDTM, # origin
TRTEDTM = TRT02EDTM

File: inst/metadata/ADSL.yml (hunk 1 above)
Comment (JiaLiu0001) [reply inst/metadata/ADSL.yml:120]: Previously, we don't have type in TRTEDTM and TRTSDTM. It should be fine to leave it alone.

Comment (Melkiades) [reply inst/metadata/ADSL.yml]: Weird... I think it should have been there

File: R/radsl.R [hunk 3]
```diff
@@ -273,7 +273,7 @@ radsl <- function(N = 400, # nolint
   # apply metadata
   if (is.null(with_trt02)) { # nolint
     ADSL <- apply_metadata(ADSL, "metadata/ADSL.yml", FALSE) # nolint
-  } else {
+  }else{# nolint
```
Comment (Melkiades) [inline R/radsl.R]: `# nolint` should be used only for exceptions in naming conventions such as `ADSL` (we use `lower_snake_case`). Here, it should be `} else {`. Other standards like these can be found in the [dev-docs](https://nest.pages.roche.com/docs/dev-guide/coding/).

Comment (JiaLiu0001) [reply R/radsl.R]: Thanks @Melkiades The code format is updated

**Review by shajoezhu — APPROVED** [review]
very nice! Thanks @JiaLiu0001 
lgtm

**Pull request merged**